src/components/Hero/Hero.test.tsx

```diff
@@ -8,202 +8,481 @@ import { HeadingLevels } from "../Heading/HeadingTypes";
+        <Hero
+          heroType={HeroTypes.Primary}
+          heading={
+            <Heading
+              level={HeadingLevels.One}
+              id="a11y-hero"
+              text="Hero Primary"
+            />
+          }
+          subHeaderText="Example Subtitle"
+          backgroundImageSrc="https://placeimg.com/1600/800/arch"
+        />
+      );
+      expect(await axe(container)).toHaveNoViolations();
+    });
+
+    it("passes for type Secondary", async () => {
+      const { container } = render(
+        <Hero
+          heroType={HeroTypes.Secondary}
+          heading={
+            <Heading
+              level={HeadingLevels.One}
+              id="a11y-hero"
+              text="Hero Secondary"
+            />
+          }
+          subHeaderText={subHeaderText}
+          image={image}
+        />
+      );
+      expect(await axe(container)).toHaveNoViolations();
+    });
+
+    it("passes for type Tertiary", async () => {
+      const { container } = render(
+        <Hero
+          heroType={HeroTypes.Tertiary}
+          heading={
+            <Heading
+              level={HeadingLevels.One}
+              id="a11y-hero"
+              text="Hero Tertiary"
+            />
+          }
+          subHeaderText={otherSubHeaderText}
+        />
+      );
+      expect(await axe(container)).toHaveNoViolations();
+    });
+
+    it("passes for type Campaign", async () => {
+      const { container } = render(
+        <Hero
+          heroType={HeroTypes.Campaign}
+          heading={
+            <Heading
+              level={HeadingLevels.One}
+              id="a11y-hero"
+              text="Hero Campaign"
+            />
+          }
+          subHeaderText={otherSubHeaderText}
+          backgroundImageSrc="https://placeimg.com/2400/800/nature/grayscale"
+          image={image}
+        />
+      );
+      expect(await axe(container)).toHaveNoViolations();
+    });
+
+    it("passes for type FiftyFifty", async () => {
+      const { container } = render(
+        <Hero
+          heroType={HeroTypes.FiftyFifty}
+          subHeaderText={otherSubHeaderText}
+          image={image}
+        />
+      );
+      expect(await axe(container)).toHaveNoViolations();
+    });
   });
 
-  it("renders Primary Hero with background image", () => {
-    const utils = render(
+  it("renders Primary Hero", () => {
+    render(
       <Hero
         heroType={HeroTypes.Primary}
         heading={
           <Heading
             level={HeadingLevels.One}
             id="primary-hero"
             text="Hero Primary"
-            blockName="hero"
           />
         }
+        subHeaderText="Example Subtitle"
         backgroundImageSrc="https://placeimg.com/1600/800/arch"
       />
     );
-    expect(utils.container.querySelector(".hero--primary")).toBeInTheDocument();
-    expect(utils.container.querySelector(".hero")).toHaveAttribute(
+
+    expect(screen.getByText("Hero Primary")).toBeInTheDocument();
```

Comment:
Right, this type of test doesn't help and it's not class-based anymore. The snapshot tests don't readily work because of Chakra's random id/classname values. So we'll have to fix that in order to get proper styling testing. That may be fixed in Chakra version 1.6.8 so I'll try that then.

Reply:
This is related to another question I had.  When we used SCSS, it was nice to see where the styles were coming from and it helped with debugging and adjusting styles.  With the Chakra CSS bundles, there is no way to tell where certain styles are being inherited from and how the CSS hierarchy is structured.  Are you aware of a config setting that would toggle bundling?  And maybe leave some type of trail to see which data object is producing certain styles?